### PR TITLE
Format all files in a repo

### PR DESCRIPTION
**Pull request by fangyi-zhou**
Use `git ls-files *.py` for git repos and `hg locate *.py` for mercurial repos to gather all python files when looking for files. This addresses the issue that when you do `black .` under a virtualenv, black starts to format files in the site-package in the virtualenv because it doesn't honour `.gitignore`. This proposed approach can honour `.gitignore` correctly.

### CODE REVIEW

File: black.py [hunk 1]
```diff
@@ -3172,6 +3183,30 @@ def get_imports_from_children(children: List[LN]) -> Generator[str, None, None]:
...
 
 
+def gen_python_files_via_vcs(root: Path) -> Iterator[Path]:
+    """Generate all python files in a repo under VCS
+    using `git ls-files` for Git Repos and `hg locate` for Mercurial Repos
+
+    `include` and `exclude` are not considered.
+    """
+    cwd = os.getcwd()
+    os.chdir(root)
```
Comment (JelleZijlstra) [inline black.py]: Better to use the `cwd=` argument to subprocess.run so you don't have to chdir.

**Comment by fangyi-zhou** [comment]
Thanks for the comments. The code is not yet complete and this is a simple prototype

Do you think this feature is going to useful?

Fangyi

— sent from a mobile device, apologies for typos

**Comment by zsol** [comment]
I would build this logic on top of black, not inside it. Separation of concerns, and all that

**Comment by fangyi-zhou** [comment]
Hi Zsolt

Thanks for the comments. So this PR should be closed?

Fangyi

— sent from a mobile device, apologies for typos

**Comment by zsol** [comment]
Yes

**Pull request closed without merging**